Charmhub requirements

Gleland opened the pull request:
add charmcraft and metadata changes to build with charmcraft pack successfully

Gleland committed:
Update metadata.yaml to support focal

Gleland committed:
Add charmcraft.yaml

satisfy requirements to build with charmcraft on launchpad.

afreiberger commented on charmcraft.yaml:
```diff
@@ -0,0 +1,18 @@
+              - amd64
+parts:
+  charm:
+    charm-python-packages: [setuptools < 58]
```
why is this necessary? I don't see jinja2 or charmhelpers in requirements.txt

Gleland replied:
good catch, thanks! i've removed this, and verified that it doesn't cause an issue when I build locally.

Gleland committed:
Remove setuptools version pinning

As discussed in [PR#1](https://github.com/canonical/charm-simple-streams/pull/1), this is not required.

afreiberger commented on metadata.yaml:
```diff
@@ -6,7 +6,7 @@ description: |
 summary: |
   Simple Streams repository
-series: [bionic]
+series: [bionic,focal]
```
Would you mind terribly using a list indent or add a space?

Gleland replied:
agreed! thanks for catching this.

afreiberger commented on charmcraft.yaml:
```diff
@@ -0,0 +1,15 @@
+    - build-on:
+        - name: ubuntu
+          channel: "20.04"
+          architectures: ["amd64"]
```
Not sure if I've approved this before, but we should be consistent in the same yaml file whether we use list indents or list notation.

Gleland replied:
agreed! thanks, this has been updated.

Gleland committed:
update indentation and list usage for consistency

Gleland committed:
Update indenation and list usage to be consistent.

afreiberger reviewed (approved):
LGTM, thanks for taking the extra time to clear that up!
